File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Syntax/BaseSyntaxExtensions.cs
using Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax;
using Microsoft.VisualBasic.CompilerServices;

namespace Microsoft.CodeAnalysis.VisualBasic.Syntax
{
	[StandardModule]
	internal sealed class BaseSyntaxExtensions
	{
		internal static Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode ToGreen(this Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode node)
		{
			return node;
		}

		internal static Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode ToGreen(this VisualBasicSyntaxNode node)
		{
			return node?.VbGreen;
		}

		internal static SyntaxNode ToRed(this Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode node)
		{
			return node?.CreateRed(null, 0);
		}

		internal static VisualBasicSyntaxNode ToRed(this VisualBasicSyntaxNode node)
		{
			return node;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Syntax/BaseSyntaxExtensions.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic
Imports Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Runtime.CompilerServices

Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax
	Friend Module BaseSyntaxExtensions
		<Extension>
		Friend Function ToGreen(ByVal node As Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode) As Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode
			Return node
		End Function

		<Extension>
		Friend Function ToGreen(ByVal node As Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode) As Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode
			If (node Is Nothing) Then
				Return Nothing
			End If
			Return node.VbGreen
		End Function

		<Extension>
		Friend Function ToRed(ByVal node As Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.VisualBasicSyntaxNode) As Microsoft.CodeAnalysis.SyntaxNode
			Dim syntaxNode As Microsoft.CodeAnalysis.SyntaxNode
			If (node IsNot Nothing) Then
				syntaxNode = node.CreateRed(Nothing, 0)
			Else
				syntaxNode = Nothing
			End If
			Return syntaxNode
		End Function

		<Extension>
		Friend Function ToRed(ByVal node As Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode) As Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode
			Return node
		End Function
	End Module
End Namespace